xapian-applications/omega/date.cc

```diff
@@ -144,7 +147,19 @@ date_range_filter(const string & date_start, const string & date_end,
 {
     int y1, m1, d1, y2, m2, d2;
     if (!date_span.empty()) {
-	time_t secs = atoi(date_span.c_str()) * (24 * 60 * 60);
+	errno = 0;
+	long long str_to_longlong = strtoll(date_span.c_str(), NULL, 0);
+	if (errno == ERANGE) {
+	    switch (str_to_longlong) {
+		case LLONG_MIN:
+		    throw (string("Date span underflow:"));
+		case LLONG_MAX:
+		    throw (string("Date span overflow:"));
+	    }
+	}
+	if (time_t(str_to_longlong) > (numeric_limits<time_t>::max() / (24 * 60 * 60)))
```

Comment:
This check is incorrect for 32-bit time_t - e.g. when `str_to_longlong = 4294967296` it evaluates to `0 > 24855`.